packages/devtools_app_shared/lib/src/service/service_extension_manager.dart

```diff
@@ -254,68 +254,79 @@ final class ServiceExtensionManager with DisposerMixin {
     }
     _hasServiceExtension(name).value = true;
 
-    if (_enabledServiceExtensions.containsKey(name)) {
+    final enabledServiceExtension = _enabledServiceExtensions[name];
+    if (enabledServiceExtension != null) {
       // Restore any previously enabled states by calling their service
       // extension. This will restore extension states on the device after a hot
       // restart. [_enabledServiceExtensions] will be empty on page refresh or
       // initial start.
       try {
-        return await _callServiceExtension(
+        final called = await _callServiceExtension(
           name,
-          _enabledServiceExtensions[name]!.value,
+          enabledServiceExtension.value,
         );
+        if (called) {
```

Comment:
can you add a comment explaining why we are only doing this if the extension was called?

Reply:
Done.